inference/core/workflows/execution_engine/entities/base.py

```diff
@@ -259,3 +272,20 @@ def to_inference_format(self, numpy_preferred: bool = False) -> Dict[str, Any]:
         if self._base64_image:
             return {"type": "base64", "value": self.base64_image}
         return {"type": "numpy_object", "value": self.numpy_image}
+
+
+class VideoMetadata(BaseModel):
+    video_identifier: str = Field(
+        description="Identifier string for video. To be treated as opaque."
+    )
+    frame_number: int
+    frame_timestamp: datetime
```

Comment:
I'd vote for timestamp to be expressed as float - seconds (including milliseconds part)
User can calculate datetime of each frame based on file creation date and seconds as offset.

Reply:
lets talk about that on call